### ISSUE TITLE
RegisterCatalogue: add description and engineering unit

### ISSUE THREAD

**Issue by mhier** [ChimeraTK/DeviceAccess#42]
(Pay attention to issue #41 - it might not be a good idea to work on these issues in parallel)

Add a description and an engineering unit field for each register to the catalogue.

NumericAddressedBackends should read the description from the optional comment field (everything after the # character of a valid register entry). The unit is currently not foreseen in the map file format - a convention should be used to extract it from the comment field, e.g. comment starts with "[<unit>]". If not found, no unit is added. If the unit token is found, it should not be part of the description in the catalogue.

DoD:
- [ ] Add fields to the RegisterInfo class
- [ ] Extend map file parser for NumericAddressedBackend to read description and unit from comment field.
- [ ] Add test for the NumericAddressedBackend parser extension
- [ ] Add user documentation: a description of the map file format currently seems to be completely missing, so add it and include the new description and unit feature

**Comment by killenb** [comment]
This task has to be split. Also non-numerically addressed backends need this feature (e.g. logical name mapping #83). Split the design and API-based tests (manually fill the catalogue) into a separate ticket. The extension of the map file needs to be synchronised with the firmware group and should be handled in this ticket.
